MODULES.md update (#1842)
epiphany-platform/epiphany: docs/home/howto/MODULES.md

## Introduction

> In version 0.8 of epiphany we are introducing modules. Modularisation of epiphany environment will result in:

Reviewer: ```suggestion
In version 0.8 of epiphany we are introducing modules. Modularization of epiphany environment will result in:
```

Author: I think is the same. UK:s, US:z. But sure, I can change it.

Author: I think it's the same. UK:s, US:z. But sure, I can change it.

> In version 0.8 of Epiphany we are introducing modules. Modularization of Epiphany environment will result in:

Reviewer 2: Maybe:
In version 0.8 of Epiphany we introduce modules

Author: Or even 'introduced' :)

* faster and more focused release cycle.

Those and multiple other factors (eg.: readability, reliability) influence this direction of changes.

## User point of view

From a user point of view, there will be no significant changes in the nearest future as it will be still possible to install Epiphany "classic way" so with a single `epicli` configuration using a whole codebase as a monolith.

For those who want to play with new features, or will need newly introduced possibilities, there will be a short transition period which we consider as a kind of "preview stage". In this period there will be a need to run each module separately by hand in the following order:

* moduleA init
* moduleA plan
* moduleA apply
* moduleB init
* moduleB plan
* moduleB apply
* ...

Init, plan and apply phases explanation you'll find in next sections of this document. Main point is that dependent modules have to be executed one after another during this what we called "preview stage". Later, with next releases there will be separate mechanism introduced to orchestrate modules dependencies and their consecutive execution.

## New scenarios

> In 0.8 we introduce the possibility to use AKS or EKS as Kubernetes providers. That is introduced with modules mechanism, so we introduced the first four modules:

Reviewer 2: introduce synonyms, maybe like this?
In 0.8 we offer the possibility to use AKS or EKS as Kubernetes providers. That is introduced with modules mechanism, so we launched the first four modules:

Author: :+1:

* [Azure Basic Infrastructure](https://github.com/epiphany-platform/m-azure-basic-infrastructure) (AzBI) module
* [Azure AKS](https://github.com/epiphany-platform/m-azure-kubernetes-service) (AzKS) module
* [AWS Basic Infrastructure](https://github.com/epiphany-platform/m-aws-basic-infrastructure) (AwsBI) module
* [AWS EKS](https://github.com/epiphany-platform/m-aws-kubernetes-service) (AwsKS) module

> Those 4 modules with the Classic Epiphany used with `any` provider allow change of on-prem Kubernetes cluster with managed Kubernetes services.

Reviewer 3: ```suggestion
Those 4 modules together with classic Epiphany used with `any` provider allow replacing of on-prem Kubernetes cluster with managed Kubernetes services.
```

As it might be already visible there are 2 paths provided:

* Azure related, using AzBI and AzKS modules,
* AWS related, using AwsBI and AwsKS modules.

> Those "... Basic Infrastructure" modules are responsible to provide basic cloud resources (eg.: resource groups, virtual networks, subnets, virtual machines, network security rules, routing, ect.) which will be used by next modules. So in this case, those are "... KS modules" meant to provide managed Kubernetes services. They use resources provided by basic infrastructure modules (eg.: subnets or resource groups) and instantiate managed Kubernetes services provided by cloud providers. The last element in both those cloud provider related paths is Classic Epiphany installed on top of resources provided by those modules using `any` provider.

Reviewer 3: ```suggestion
Those "... Basic Infrastructure" modules are responsible to provide basic cloud resources (eg.: resource groups, virtual networks, subnets, virtual machines, network security rules, routing, ect.) which will be used by next modules. So in this case, those are "... KS modules" meant to provide managed Kubernetes services. They use resources provided by basic infrastructure modules (eg.: subnets or resource groups) and instantiate managed Kubernetes services provided by cloud providers. The last element in both those cloud provider related paths is classic Epiphany installed on top of resources provided by those modules using `any` provider.
```

  specification:
    name: your-cluster-name # <----- make unified with other places and build directory name
    admin_user:
      name: operations # <----- make sure os-user is correct

>       key_path: /tmp/shared/vms_rsa # <----- use generated keyfile

Reviewer 3: ```suggestion
      key_path: /tmp/shared/vms_rsa # <----- use generated key file
```

    cloud:

>       k8s_as_cloud_service: true # <----- make sure that flag is set, as it indicates usage of AKS service

Reviewer 3: ```suggestion
      k8s_as_cloud_service: true # <----- make sure that flag is set, as it indicates usage of a managed Kubernetes service
```
